apps/cli-daemon/src/app/workspace/workspace.service.ts

```diff
@@ -0,0 +1,87 @@
+import { NodeJsSyncHost } from '@angular-devkit/core/node';
+import {
+  createWorkspaceHost,
+  // writeWorkspace,
+  readWorkspace as devKitReadWorkspace,
+  ProjectDefinition,
+  TargetDefinition,
+  WorkspaceDefinition,
+} from '@angular-devkit/core/src/workspace';
+import {
+  BadRequestException,
+  Injectable,
+  InternalServerErrorException,
+  Logger,
+  NotFoundException,
+} from '@nestjs/common';
+
+import { SessionService } from '../session/session.service';
+
+export const BAD_PATH_EXCEPTION =
+  'Unable to determine format for workspace path.';
+export const NOT_ANGULAR_WORKSPACE =
+  'Unable to locate a workspace file for workspace path. Are you missing an `angular.json`' +
+  ' or `.angular.json` file?';
+
+@Injectable()
+export class WorkspaceService {
+  private readonly logger = new Logger(WorkspaceService.name);
+
+  constructor(private readonly sessionService: SessionService) {}
+
+  async readWorkspace(path: string): Promise<WorkspaceDefinition> {
+    try {
+      return (
+        await devKitReadWorkspace(
+          path,
+          createWorkspaceHost(new NodeJsSyncHost())
+        )
+      ).workspace;
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (err: any) {
+      const errorMessage = err['message'];
+      switch (errorMessage) {
+        case BAD_PATH_EXCEPTION:
+          throw new BadRequestException(BAD_PATH_EXCEPTION);
+        case NOT_ANGULAR_WORKSPACE:
+          throw new NotFoundException('Angular workspace not found.');
```

Comment:
also put in a variable

Reply:
Why? It's our message used in only one place, not the one I took from the devkit.

Reply:
to keep convention with the BAD_PATH_EXCEPTION